pkg/middleware/static.go

```diff
@@ -0,0 +1,37 @@
+package middleware
+
+import (
+	"net/http"
+	"strings"
+)
+
+// Static is a middleware that serves static assets.
+func Static(root string, fs http.FileSystem) func(http.Handler) http.Handler {
+	if !strings.HasSuffix(root, "/") {
+		root = root + "/"
+	}
+
+	static := http.StripPrefix(
+		root,
+		http.FileServer(
+			fs,
+		),
+	)
+
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// serve the static assets for the identifier web app
+			if strings.HasPrefix(r.URL.Path, "/signin/v1/static/") {
+				if strings.HasSuffix(r.URL.Path, "/") {
```

Comment:
there is a middle ware which strips slashes ... can we use that in here?
https://godoc.org/github.com/pressly/chi/middleware#StripSlashes

Reply:
The idea is that we want to prevent rendering a file listing ... Maybe it is possible to tell the assets not to render them. This also works, but just for urls that end in /

Reply:
I thought of adding an option to the assets that would prevent listing folders by returning ErrNotExist ... but that would mix assets with functionality that uses the assets. I think the check if a folder is requested should go into the static middleware.

...

Ok I stepped through this. The http.Filesystem redirects directories to a url that ends in `/` so `if strings.HasSuffix(r.URL.Path, "/")`properly prevents dir listings. We explicitly want to check if a `/` is there and not get rid of it ... so all good. This works as intended.